examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py

```diff
@@ -77,10 +77,38 @@ def check_keys_for_asset_keys(
 
 
 def build_airflow_polling_sensor_defs(
-    airflow_data: AirflowDefinitionsData,
+    *,
+    mapped_defs: Definitions,
+    airflow_instance: AirflowInstance,
     event_transformer_fn: Optional[DagsterEventTransformerFn],
     minimum_interval_seconds: int = DEFAULT_AIRFLOW_SENSOR_INTERVAL_SECONDS,
 ) -> Definitions:
+    """Builds a sensor that polls Airflow for new dag runs and emits materializations for each asset.
+
```

Comment:
The constructed sensor polls the airflow instance for activity inserts materializations (and other events) into Dagster's event log. The sensor decides what dags and tasks to monitor by inspecting the _metadata_ of the passed definitions. This metadata is typically added by `dag_defs` and `task_defs` at definition construction time.

Once the metadata is applied a _mapping_ exists between the airflow entity and the dagster entity. A "mapped" definitions objects this metadata exists on those definitions, and the sensor depends on their existence to function properly.

Reply:
^--- example top-level language we can put somewere